pkg/apis/core/validation/cloudprofile.go

```diff
@@ -44,11 +45,65 @@ func ValidateCloudProfileUpdate(newProfile, oldProfile *core.CloudProfile) field
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProfile.ObjectMeta, &oldProfile.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateCloudProfileSpecUpdate(&newProfile.Spec, &oldProfile.Spec, field.NewPath("spec"))...)
 	allErrs = append(allErrs, ValidateCloudProfile(newProfile)...)
 
 	return allErrs
 }
 
+// ValidateCloudProfileSpecUpdate validates the spec update of a CloudProfile
+func ValidateCloudProfileSpecUpdate(new *core.CloudProfileSpec, old *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateCloudProfileVersionsUpdate(new.Kubernetes.Versions, old.Kubernetes.Versions, fldPath.Child("kubernetes", "versions"))...)
+
+	for index, _ := range old.MachineImages {
+		if len(new.MachineImages) - 1  >= index {
+			allErrs = append(allErrs, validateCloudProfileVersionsUpdate(new.MachineImages[index].Versions, old.MachineImages[index].Versions, fldPath.Child("machineImages").Index(index).Child("versions"))...)
+		}
+	}
+
+	return allErrs
+}
+
+// ValidateCloudProfileAddedVersions validates versions added to the CloudProfile
+func ValidateCloudProfileAddedVersions(versions []core.ExpirableVersion, addedVersions map[int]core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for index, version := range addedVersions {
+		// Do not allow adding addedVersions with expiration date in the past
+		if  version.ExpirationDate != nil && version.ExpirationDate.Time.UTC().Before(time.Now()) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(index), version.Version, fmt.Sprintf("unable to add version '%s'. Adding a version with expiration date in the past is not allowed", version.Version)))
+			continue
+		}
+
+		if version.Classification != nil && *version.Classification == core.ClassificationSupported {
+			currentSemVer, err := semver.NewVersion(version.Version)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(index), version.Version, fmt.Sprintf("unable to add version '%s'. Version should be semVer compatible", version.Version)))
```

Comment:
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Index(index), version.Version, fmt.Sprintf("unable to add version '%s'. Version must be a semantic version", version.Version)))
```

Reply:
Why is this check only performed if the classification is supported?

Reply:
This is only to obtain the semantic version and is already seperatly checked here for machine versions: https://github.com/gardener/gardener/blob/deb59ec01131a39b18be35267466590219027078/pkg/apis/core/validation/cloudprofile.go#L255-L254  (will adjust error message here too) and here via regex for Kubernetes versions: https://github.com/gardener/gardener/blob/deb59ec01131a39b18be35267466590219027078/pkg/apis/core/validation/cloudprofile.go#L154-L153